Formula/gdb.rb

```diff
@@ -48,8 +50,13 @@ def install
     args << "--with-guile" if build.with? "guile@2.0"
     args << "--enable-targets=all" if build.with? "all-targets"
 
-    if build.with? "python"
+    if build.with?("python") && build.with?("python3")
+      odie "Options --with-python and --with-python3 are mutually exclusive."
+    elsif build.with?("python")
       args << "--with-python=#{Formula["python"].opt_libexec}/bin"
+    elsif build.with?("python3")
+      args << "--with-python=#{Formula["python3"].opt_bin}/python3"
+      ENV.append "CPPFLAGS", "-I#{Formula["python"].opt_libexec}"
```

Comment:
This is a bit strange.

Reply:
Yes, I found it strange too. Unfortunately, providing the executable to GDB isn't sufficient, because it looks `python.h` and can't find it (since in the Homebrew Python 3 formula, it isn't linked in the `opt_libexec` directory like Python 2). So we have to give it the extra path to `python.h` via `CPPFLAGS`.

Reply:
Ad providing the executable path isn't sufficient because it looks for Python < 3.

Reply:
Oops, I now see what you mean. Fixing.

Reply:
Fixed.